### PR TITLE
Remove unused `memberships` Map and pattern

### PR DESCRIPTION
**Pull request by kevinhaube**
This PR removes the idea of having more than one group membership within our authentication service. All users are expected to be in **one group only**, and for admins, we will default to their admin group over any non-admin group. 

Also in this PR, we refactor our test suite for `useOktaMemberships` to decouple it from the idea of selecting memberships from the removed `memberships` map.

Tangential work to #7923 to simplify the addition of services in a memberhsip.

Test Steps:
1. Ensure you can log in, and still operate within the application
2. Observe your membership state via React Dev Tools (find SessionProvider and view its state) or localStorage to ensure no `memberships` map was created/populated
3. Run the full test suite to ensure no failing tests

## Changes
- Removes `memberships` map 
- Refactors `useOktaMemberships` tests

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/hooks/UseOktaMemberships.test.ts [hunk 1]
```diff
@@ -19,33 +19,6 @@ const mockStoreSessionMembershipState = jest.fn();
 const mockStoreOrganizationOverride = jest.fn();
 const mockUpdateApiSessions = jest.fn();
 
-const fakeMemberships = new Map([
```
Comment (stephenkao) [inline frontend-react/src/hooks/UseOktaMemberships.test.ts:22]: 🔥 🔥 🔥 bye bye

File: frontend-react/src/hooks/UseOktaMemberships.ts [hunk 2]
```diff
@@ -96,46 +98,33 @@ export const getSettingsFromOrganization = (
...
+    const claims = toRSClaims(token.claims);
+    // Check if we have any organization claims
+    if (!claims?.organization?.length) {
         return omit(defaultState, "initialized");
     }
-    // Get defaults
-    const [first] = claimData.keys();
-    const active = claimData.get(first);
+    // If we're admins, set that always, else set default to first in org claim array
+    const adminIndex = claims.organization.indexOf(PRIME_ADMINS);
```
Comment (stephenkao) [inline frontend-react/src/hooks/UseOktaMemberships.ts:121]: Smalllll point, but we could get rid of the index middleman and just return it directly:

```js
return {
    activeMembership: claims.organization.find((org) => org === PRIME_ADMINS) || claims.organization[0]
}
```

**Review by stephenkao — APPROVED** [review]
Looks great!  Thanks for taking the time for this cleanup!  🧹 🧹 🧹

File: frontend-react/src/hooks/UseOktaMemberships.ts (hunk 2 above)
Comment (kevinhaube) [reply frontend-react/src/hooks/UseOktaMemberships.ts]: Niiiice, love these little in-line tricks! Patched up

**Pull request merged**